Make sure to check in correct path for a valid init. This was failing if
/sbin/init is a symlink to /lib/systemd/systemd in voidlinux.

--- modules.d/99base/init.sh.orig	2012-05-28 15:28:06.363878326 +0200
+++ modules.d/99base/init.sh	2012-05-28 15:29:03.361485893 +0200
@@ -234,7 +234,7 @@ for i in "$(getarg real_init=)" "$(getar
     [ -n "$i" ] || continue
 
     __p=$(readlink -f "${NEWROOT}/${i}")
-    if [ -x "$__p" ]; then
+    if [ -x "$__p" -o -x "${NEWROOT}/${__p}" ]; then
         INIT="$i"
         break
     fi
